src/cow-react/modules/limitOrders/hooks/useLimitOrdersTradeState.ts

```diff
@@ -24,9 +25,10 @@ export interface LimitOrdersTradeState {
 export function useLimitOrdersTradeState(): LimitOrdersTradeState {
   const { account } = useWeb3React()
   const state = useAtomValue(limitOrdersAtom)
+  const settingsState = useAtomValue(limitOrdersSettingsAtom)
 
   const recipient = state.recipient
-  const deadline = state.deadline
+  const deadline = settingsState.customDeadline ? settingsState.customDeadline - Date.now() : settingsState.deadline
```

Comment:
this is a bit strange, right? 

if I understand, you deduct the "now" so later when you add back the "now" you get the right absolute time. 

However, you won't get exactly the same time, and feels a bit artificial. 

Could it be that we try to model deadlineDuration and deadline in the same var?

Reply:
Yes, it's because `settingsState.customDeadline` is timestamp and `settingsState.deadline` is a count of milliseconds.
I `deduct` and `add` just to simplify usage of `deadline`.
Unfortunately cause of it we can't store `deadline` and `customDeadline` in the same var.
Also, we must calculate `validTo` regarding to `deadline` only at the moment when we sign a limit order.

Reply:
Would it make sense to create a hook that would have this logic and return `deadline` and maybe even `validTo`. Also for the `validTo` can we reuse this https://github.com/cowprotocol/cowswap/blob/ea734d030bb2e9392da2bcfb8d9cb2792f3f25f5/src/cow-react/utils/time.ts#L35 or is the calculation in this case different

Reply:
Thanks! Used `calculateValidTo()`

Reply:
My point was more, if we allow to define the deadline as a relative time, and as an absolute time, and this have two different behaviours, then maybe we need to have 2 different settings, or model it as something in these lines:

```js
type Deadline = AbsoluteDeadline | RelativeDeadline
interface AbsoluteDeadline {
    absoluteDeadline: Date
    kind: 'absolute'
}

interface RelativeDeadline {
    relativeDeadlineSeconds: number
    kind: 'relative'
}

const deadline1: Deadline = { kind: 'absolute', absoluteDeadline: new Date()  }
const deadline2: Deadline =  { kind: 'relative', relativeDeadlineSeconds: 123456 }
```

See:

https://www.typescriptlang.org/play?#code/C4TwDgpgBAIhCGATANgSwHbQLxQIICMBnAe2QFdgI4k1MoAfKAJQmXmFQDcqEUMIAUBkoAnAGbwAxtAIlylanzoBvAVHVR4RUhR41+ALljtBGqAGsMiIwHItc3TYEBfAUPSiJ05q3Zc9StCqZiK+HNyKtBAAyhCSxOiIhEboZAC2+BAiahqWibahbOEQTq4C8eiEwFCIvFEAjEaR-FA4yhZWtvY6lDYANJra8gFRKRAA7saUABQAlOquFVU1dfwATE2rdDhQ7XnWUDaFftz9UMfFzZixFUlG9WsAzAAsAKwAbFBlS9W1+piEVpQADaf0C9QGYKiawAugIofxCAA6MTEEQAUSkAAtpmARMIrhBZm4BGIyOhJBwElA8QSthBpgjMJt-kTdmYcupUGIoIz6Uj9q0sDg7ENHPNgmYNEtSBAkchiABzaY2WQ9aBMiC2KAAahWrKR3WGhOJZmcUFYhCCnKlMuQcoVypsLCK-n1gW1es1SIu-kJNwSSVNGlcziAA

Reply:
@anxolin thanks! I got your idea.
https://github.com/cowprotocol/cowswap/pull/1283/commits/93316df5ec6348182a73fe74755b011c15417168 - what do you think about the approach?

Now we have only one place where we calculate `deadlineTimestamp`: https://github.com/cowprotocol/cowswap/pull/1283/commits/93316df5ec6348182a73fe74755b011c15417168#diff-f9e6f83838cf4e557a8b564f81ca086e2cc023a95bf146100bf5ed5e902a42a8R34-R36

Because of it, I think adding the `kind` property would be overengineering. I don't see any cases in the future to touch these values.